test/test_pebble.py

```diff
@@ -520,53 +658,103 @@ def _services_action_async_helper(self, action, api_func, services):
         def api_func(timeout=30):
-            return self.api.start_services(['svc'], timeout=timeout)
+            return self.client.start_services(['svc'], timeout=timeout)
         self._services_action_async_helper('start', api_func, ['svc'])
 
     def test_stop_services(self):
         def api_func():
-            return self.api.stop_services(['svc'])
+            return self.client.stop_services(['svc'])
         self._services_action_helper('stop', api_func, ['svc'])
 
     def test_stop_services_async(self):
         def api_func(timeout=30):
-            return self.api.stop_services(['svc'], timeout=timeout)
+            return self.client.stop_services(['svc'], timeout=timeout)
         self._services_action_async_helper('stop', api_func, ['svc'])
 
+    def test_service_error(self):
+        self.client.responses.append({
+            "change": "70",
+            "result": None,
+            "status": "Accepted",
+            "status-code": 202,
+            "type": "async"
+        })
+        change = self.build_mock_change_dict()
+        change['err'] = 'Some kind of service error'
+        self.client.responses.append({
+            "result": change,
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        with self.assertRaises(pebble.ServiceError) as cm:
+            self.client.autostart_services()
+        self.assertEqual(cm.exception.err, 'Some kind of service error')
+        self.assertIsInstance(cm.exception.change, pebble.Change)
+        self.assertEqual(cm.exception.change.id, '70')
+
     def test_wait_change_timeout(self):
         with unittest.mock.patch('ops.pebble.time', MockTime()):
             change = self.build_mock_change_dict()
             change['ready'] = False
             for _ in range(3):
-                self.api.responses.append({
+                self.client.responses.append({
                     "result": change,
                     "status": "OK",
                     "status-code": 200,
                     "type": "sync"
                 })
 
             with self.assertRaises(TimeoutError):
-                self.api.wait_change('70', timeout=3, delay=1)
+                self.client.wait_change('70', timeout=3, delay=1)
 
-            self.assertEqual(self.api.requests, [
+            self.assertEqual(self.client.requests, [
                 ('GET', '/v1/changes/70', None, None),
                 ('GET', '/v1/changes/70', None, None),
                 ('GET', '/v1/changes/70', None, None),
             ])
+
+
+class TestSocketClient(unittest.TestCase):
+    @unittest.skipIf(sys.platform == 'win32', "Unix sockets don't work on Windows")
+    def test_socket_not_found(self):
+        client = pebble.Client(socket_path='does_not_exist')
+        with self.assertRaises(urllib.error.URLError):
+            client.get_system_info()
+
+    @unittest.skipIf(sys.platform == 'win32', "Unix sockets don't work on Windows")
```

Comment:
IRC, there was some issue about platform and using win32, but when I dug the only thing I found was that linux2 was renamed to linux back in python 3.3 (because Linus changed something that would have made it linux3 according to the kernel version.)
I did confirm that python3.7 in 64 bit mode does, indeed, report win32.

Reply:
Yep, that's my take too -- `win32` seems to be the documented, correct value for `sys.platform` on Windows, whether 64-bit or not.